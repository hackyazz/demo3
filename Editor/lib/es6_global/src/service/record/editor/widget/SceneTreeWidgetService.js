


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
