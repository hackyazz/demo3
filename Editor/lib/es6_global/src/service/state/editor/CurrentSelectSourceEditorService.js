


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
