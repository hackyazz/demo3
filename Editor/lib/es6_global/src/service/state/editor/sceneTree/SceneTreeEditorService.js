

import * as Js_option from "../../../../../../../node_modules/bs-platform/lib/es6/js_option.js";
import * as ImmutableSparseMapService$WonderCommonlib from "../../../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ImmutableSparseMapService.js";
import * as IsShowChildrenSceneTreeService$WonderEditor from "../../../record/editor/sceneTree/IsShowChildrenSceneTreeService.js";
import * as CurrentSceneTreeNodeSceneTreeService$WonderEditor from "../../../record/editor/sceneTree/CurrentSceneTreeNodeSceneTreeService.js";

function unsafeGetCurrentSceneTreeNode(editorState) {
  return CurrentSceneTreeNodeSceneTreeService$WonderEditor.unsafeGetCurrentSceneTreeNode(editorState[/* sceneTreeRecord */1]);
}

function getCurrentSceneTreeNode(editorState) {
  return CurrentSceneTreeNodeSceneTreeService$WonderEditor.getCurrentSceneTreeNode(editorState[/* sceneTreeRecord */1]);
}

function setCurrentSceneTreeNode(gameObject, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */CurrentSceneTreeNodeSceneTreeService$WonderEditor.setCurrentSceneTreeNode(gameObject, editorState[/* sceneTreeRecord */1]),
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function hasCurrentSceneTreeNode(editorState) {
  return Js_option.isSome(CurrentSceneTreeNodeSceneTreeService$WonderEditor.getCurrentSceneTreeNode(editorState[/* sceneTreeRecord */1]));
}

function clearCurrentSceneTreeNode(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */CurrentSceneTreeNodeSceneTreeService$WonderEditor.clearCurrentSceneTreeNode(editorState[/* sceneTreeRecord */1]),
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function getIsShowChildrenMap(param) {
  return IsShowChildrenSceneTreeService$WonderEditor.getIsShowChildrenMap(param[/* sceneTreeRecord */1]);
}

function getDefaultIsShowChildren(param) {
  return false;
}

function getIsShowChildern(gameObject, sceneGameObject, param) {
  var match = gameObject === sceneGameObject;
  if (match) {
    return true;
  } else {
    var match$1 = ImmutableSparseMapService$WonderCommonlib.get(gameObject, IsShowChildrenSceneTreeService$WonderEditor.getIsShowChildrenMap(param[/* sceneTreeRecord */1]));
    if (match$1 !== undefined) {
      return match$1;
    } else {
      return false;
    }
  }
}

function setIsShowChildren(gameObject, isShowChildren, editorState) {
  var sceneTreeRecord = editorState[/* sceneTreeRecord */1];
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord : record */[
            /* currentSceneTreeNode */sceneTreeRecord[/* currentSceneTreeNode */0],
            /* isShowChildrenMap */ImmutableSparseMapService$WonderCommonlib.set(gameObject, isShowChildren, IsShowChildrenSceneTreeService$WonderEditor.getIsShowChildrenMap(sceneTreeRecord))
          ],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function removeIsShowChildren(gameObject, editorState) {
  var sceneTreeRecord = editorState[/* sceneTreeRecord */1];
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord : record */[
            /* currentSceneTreeNode */sceneTreeRecord[/* currentSceneTreeNode */0],
            /* isShowChildrenMap */ImmutableSparseMapService$WonderCommonlib.deleteVal(gameObject, IsShowChildrenSceneTreeService$WonderEditor.getIsShowChildrenMap(sceneTreeRecord))
          ],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  unsafeGetCurrentSceneTreeNode ,
  getCurrentSceneTreeNode ,
  setCurrentSceneTreeNode ,
  hasCurrentSceneTreeNode ,
  clearCurrentSceneTreeNode ,
  getIsShowChildrenMap ,
  getDefaultIsShowChildren ,
  getIsShowChildern ,
  setIsShowChildren ,
  removeIsShowChildren ,
  
}
/* CurrentSceneTreeNodeSceneTreeService-WonderEditor Not a pure module */
