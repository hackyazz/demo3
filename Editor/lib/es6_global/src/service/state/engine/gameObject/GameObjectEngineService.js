

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Log$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as Contract$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Contract.js";
import * as GameObjectAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/GameObjectAPI.js";
import * as ArrayService$WonderCommonlib from "../../../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ArrayService.js";
import * as StateEditorService$WonderEditor from "../../editor/StateEditorService.js";
import * as IsRootGameObjectMainService$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/gameObject/IsRootGameObjectMainService.js";
import * as MeshRendererEngineService$WonderEditor from "../MeshRendererEngineService.js";
import * as BasicCameraViewEngineService$WonderEditor from "../camera/BasicCameraViewEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "./GameObjectComponentEngineService.js";
import * as HierarchyGameObjectEngineService$WonderEditor from "./HierarchyGameObjectEngineService.js";

function cloneGameObject(gameObject, count, isShareMaterial, engineState) {
  var match = GameObjectAPI$Wonderjs.cloneGameObject(gameObject, count, isShareMaterial, engineState);
  return /* tuple */[
          match[1],
          match[0]
        ];
}

function setGameObjectName(name, gameObject, engineState) {
  return GameObjectAPI$Wonderjs.setGameObjectName(gameObject, name, engineState);
}

var getGameObjectIsRoot = IsRootGameObjectMainService$Wonderjs.getIsRoot;

function _getAllComponents(allGameObjects, param, engineState) {
  var unsafeGetComponentFunc = param[1];
  var hasComponentFunc = param[0];
  return ArrayService$WonderCommonlib.removeDuplicateItems(allGameObjects.filter((function (gameObject) {
                      return Curry._2(hasComponentFunc, gameObject, engineState);
                    })).map((function (gameObject) {
                    return Curry._2(unsafeGetComponentFunc, gameObject, engineState);
                  })));
}

function getAllBasicMaterials(allGameObjects, engineState) {
  return _getAllComponents(allGameObjects, /* tuple */[
              GameObjectComponentEngineService$WonderEditor.hasBasicMaterialComponent,
              GameObjectComponentEngineService$WonderEditor.unsafeGetBasicMaterialComponent
            ], engineState);
}

function getAllLightMaterials(allGameObjects, engineState) {
  return _getAllComponents(allGameObjects, /* tuple */[
              GameObjectComponentEngineService$WonderEditor.hasLightMaterialComponent,
              GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent
            ], engineState);
}

function getAllArcballCameraControllers(allGameObjects, engineState) {
  return _getAllComponents(allGameObjects, /* tuple */[
              GameObjectComponentEngineService$WonderEditor.hasArcballCameraControllerComponent,
              GameObjectComponentEngineService$WonderEditor.unsafeGetArcballCameraControllerComponent
            ], engineState);
}

function getAllDirectionLights(allGameObjects, engineState) {
  return _getAllComponents(allGameObjects, /* tuple */[
              GameObjectComponentEngineService$WonderEditor.hasDirectionLightComponent,
              GameObjectComponentEngineService$WonderEditor.unsafeGetDirectionLightComponent
            ], engineState);
}

function getAllPointLights(allGameObjects, engineState) {
  return _getAllComponents(allGameObjects, /* tuple */[
              GameObjectComponentEngineService$WonderEditor.hasPointLightComponent,
              GameObjectComponentEngineService$WonderEditor.unsafeGetPointLightComponent
            ], engineState);
}

function initAllGameObjects(gameObject, engineState) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (engineState, gameObject) {
                return GameObjectAPI$Wonderjs.initGameObject(gameObject, engineState);
              }), engineState, HierarchyGameObjectEngineService$WonderEditor.getAllGameObjects(gameObject, engineState));
}

function _getGameObjectActiveBasicCameraViews(gameObject, engineState) {
  return Contract$WonderLog.ensureCheck((function (activedBasicCameraViews) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("only has 0 or 1 active basicCameraView", "not"), (function (param) {
                              return Contract$WonderLog.Operators[/* <= */11](activedBasicCameraViews.length, 1);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), HierarchyGameObjectEngineService$WonderEditor.getAllGameObjects(gameObject, engineState).filter((function (gameObject) {
                        return GameObjectComponentEngineService$WonderEditor.hasBasicCameraViewComponent(gameObject, engineState);
                      })).map((function (gameObject) {
                      return GameObjectComponentEngineService$WonderEditor.unsafeGetBasicCameraViewComponent(gameObject, engineState);
                    })).filter((function (basicCameraView) {
                    return BasicCameraViewEngineService$WonderEditor.isActiveBasicCameraView(basicCameraView, engineState);
                  })));
}

function getGameObjectActiveBasicCameraView(gameObject, engineState) {
  var activeBasicCameraViews = _getGameObjectActiveBasicCameraViews(gameObject, engineState);
  var match = activeBasicCameraViews.length === 0;
  if (match) {
    return undefined;
  } else {
    return activeBasicCameraViews[0];
  }
}

function setAllGameObjectsIsRenderIfHasMeshRenderer(isRender, gameObject, engineState) {
  var _iterateGameObjectArr = function (gameObjectArr, engineState) {
    return ArrayService$WonderCommonlib.reduceOneParam((function (engineState, gameObject) {
                  var match = GameObjectComponentEngineService$WonderEditor.hasMeshRendererComponent(gameObject, engineState);
                  var engineState$1 = match ? MeshRendererEngineService$WonderEditor.setMeshRendererIsRender(GameObjectComponentEngineService$WonderEditor.unsafeGetMeshRendererComponent(gameObject, engineState), isRender, engineState) : engineState;
                  return _iterateGameObjectArr(HierarchyGameObjectEngineService$WonderEditor.getChildren(gameObject, engineState$1), engineState$1);
                }), engineState, gameObjectArr);
  };
  return _iterateGameObjectArr(/* array */[gameObject], engineState);
}

var create = GameObjectAPI$Wonderjs.createGameObject;

var initGameObject = GameObjectAPI$Wonderjs.initGameObject;

var isGameObjectAlive = GameObjectAPI$Wonderjs.isGameObjectAlive;

var disposeGameObject = GameObjectAPI$Wonderjs.disposeGameObject;

var disposeGameObjectKeepOrder = GameObjectAPI$Wonderjs.disposeGameObjectKeepOrder;

var disposeGameObjectKeepOrderRemoveGeometry = GameObjectAPI$Wonderjs.disposeGameObjectKeepOrderRemoveGeometry;

var disposeGameObjectKeepOrderRemoveGeometryRemoveMaterial = GameObjectAPI$Wonderjs.disposeGameObjectKeepOrderRemoveGeometryRemoveMaterial;

var disposeGameObjectDisposeGeometryRemoveMaterial = GameObjectAPI$Wonderjs.disposeGameObjectDisposeGeometryRemoveMaterial;

var getGameObjectName = GameObjectAPI$Wonderjs.getGameObjectName;

var unsafeGetGameObjectName = GameObjectAPI$Wonderjs.unsafeGetGameObjectName;

var unsafeGetGameObjectIsRoot = GameObjectAPI$Wonderjs.unsafeGetGameObjectIsRoot;

var setGameObjectIsRoot = GameObjectAPI$Wonderjs.setGameObjectIsRoot;

export {
  create ,
  initGameObject ,
  isGameObjectAlive ,
  disposeGameObject ,
  cloneGameObject ,
  disposeGameObjectKeepOrder ,
  disposeGameObjectKeepOrderRemoveGeometry ,
  disposeGameObjectKeepOrderRemoveGeometryRemoveMaterial ,
  disposeGameObjectDisposeGeometryRemoveMaterial ,
  getGameObjectName ,
  unsafeGetGameObjectName ,
  setGameObjectName ,
  getGameObjectIsRoot ,
  unsafeGetGameObjectIsRoot ,
  setGameObjectIsRoot ,
  _getAllComponents ,
  getAllBasicMaterials ,
  getAllLightMaterials ,
  getAllArcballCameraControllers ,
  getAllDirectionLights ,
  getAllPointLights ,
  initAllGameObjects ,
  _getGameObjectActiveBasicCameraViews ,
  getGameObjectActiveBasicCameraView ,
  setAllGameObjectsIsRenderIfHasMeshRenderer ,
  
}
/* Log-WonderLog Not a pure module */
