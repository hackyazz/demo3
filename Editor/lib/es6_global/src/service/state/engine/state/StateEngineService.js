

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
