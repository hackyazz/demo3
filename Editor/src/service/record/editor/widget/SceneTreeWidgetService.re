open WidgetType;

let getWidget = () => SceneTree;

let isWidget = startWidget =>
  switch (startWidget) {
  | None => false
  | Some(startWidget) => startWidget === getWidget()
  };