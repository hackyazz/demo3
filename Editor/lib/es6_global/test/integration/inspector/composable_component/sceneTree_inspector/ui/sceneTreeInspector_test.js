

import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../tool/TestTool.js";
import * as InspectorTool$WonderEditor from "../../../../../tool/ui/InspectorTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../tool/BuildComponentTool.js";
import * as SceneEngineService$WonderEditor from "../../../../../../src/service/state/engine/SceneEngineService.js";
import * as StateEngineService$WonderEditor from "../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as SceneTreeEditorService$WonderEditor from "../../../../../../src/service/state/editor/sceneTree/SceneTreeEditorService.js";
import * as SceneTreeInspectorTool$WonderEditor from "../tool/SceneTreeInspectorTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as GameObjectEngineService$WonderEditor from "../../../../../../src/service/state/engine/GameObjectEngineService.js";
import * as MainEditorSceneTreeTool$WonderEditor from "../../../../../unit/tool/MainEditorSceneTreeTool.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";

describe("SceneTreeInspector", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("prepare currentSelectSource is SceneTree", (function () {
                beforeEach((function () {
                        var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
                        return StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                                      return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                                    }));
                      }));
                describe("test rename gameObject", (function () {
                        beforeEach((function () {
                                return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                              }));
                        describe("test logic", (function () {
                                describe("test engine", (function () {
                                        return Wonder_jest.test("test rename gameObject ", (function () {
                                                      var newName = "gameObjectNewName";
                                                      SceneTreeInspectorTool$WonderEditor.renameGameObject(newName, undefined, undefined, undefined, /* () */0);
                                                      var partial_arg = StateLogicService$WonderEditor.getEditorState(SceneTreeEditorService$WonderEditor.unsafeGetCurrentSceneTreeNode);
                                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                            return GameObjectEngineService$WonderEditor.unsafeGetGameObjectName(partial_arg, param);
                                                                          }))), newName);
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                describe("test show Scene inspector", (function () {
                        return Wonder_jest.test("should show nothing", (function () {
                                      MainEditorSceneTreeTool$WonderEditor.Select[/* selectGameObject */0](undefined, undefined, SceneEngineService$WonderEditor.getSceneGameObject(StateEngineService$WonderEditor.unsafeGetState(/* () */0)), /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildInspectorComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), InspectorTool$WonderEditor.buildFakeAllShowComponentConfig(/* () */0)));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
