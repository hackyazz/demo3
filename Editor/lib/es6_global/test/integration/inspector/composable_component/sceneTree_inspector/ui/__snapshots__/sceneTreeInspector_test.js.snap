// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneTreeInspector prepare currentSelectSource is SceneTree test rename gameObject test rename scene ui->scene tree->scene node->name should be updated 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          gameObjectNewName
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container select-active no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Direction Light
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;

exports[`SceneTreeInspector prepare currentSelectSource is SceneTree test show Scene inspector should show component ui 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div
      className="sceneTree-name"
    >
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="scene"
          />
        </div>
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Position
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Rotation
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
          >
            Scale
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
          </div>
        </article>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <div
          className="content-btn"
        >
          <button
            className="addable-btn"
            onClick={[Function]}
          >
            Add Component
          </button>
        </div>
      </div>
    </article>
  </article>
</article>
`;
